lib/topological_inventory/orchestrator/worker.rb

```diff
@@ -63,39 +65,41 @@ def make_openshift_match_database
 
 
       ### API STUFF
-      def each_source
+      def each_source(tenant_account)
         return enum_for(:each_source) unless block_given?
-        each_resource(url_for("source_types")) do |source_type|
+        each_resource(url_for("source_types"), tenant_account) do |source_type|
           collector_definition = collector_definitions[source_type["name"]]
           next if collector_definition.nil?
-          each_resource(url_for("source_types/#{source_type["id"]}/sources")) do |source|
-            next unless endpoint = get_and_parse(url_for("sources/#{source["id"]}/endpoints"))["data"].first
-            next unless authentication = get_and_parse(url_for("authentications?resource_type=Endpoint&resource_id=#{endpoint["id"]}"))["data"].first
+          each_resource(url_for("source_types/#{source_type["id"]}/sources"), tenant_account) do |source|
+            next unless endpoint = get_and_parse(url_for("sources/#{source["id"]}/endpoints"), tenant_account)["data"].first
+            next unless authentication = get_and_parse(url_for("authentications?resource_type=Endpoint&resource_id=#{endpoint["id"]}"), tenant_account)["data"].first
             yield source, endpoint, authentication, collector_definition
           end
         end
       end
 
       def collectors_from_sources_api
         hash = {}
-        each_source do |source, endpoint, authentication, collector_definition|
-          auth = authentication_with_password(authentication["id"])
-          value = {
-            "endpoint_host"   => endpoint["host"],
-            "endpoint_path"   => endpoint["path"],
-            "endpoint_port"   => endpoint["port"].to_s,
-            "endpoint_scheme" => endpoint["scheme"],
-            "image"           => collector_definition["image"],
-            "image_namespace" => collector_definition["image_namespace"] || ENV["IMAGE_NAMESPACE"],
-            "source_id"       => source["id"],
-            "source_uid"      => source["uid"],
-            "secret"          => {
-              "password" => auth["password"],
-              "username" => auth["username"],
-            },
-          }
-          key = digest(value)
-          hash[key] = value
+        each_tenant do |tenant|
+          each_source(tenant) do |source, endpoint, authentication, collector_definition|
```

Comment:
Minor, but structurally something feels off about this...it reads funny to me that we would nest each_* in this `collectors_from_sources_api` method, as opposed to having that nesting live somewhere else, perhaps with a different name.  That is, I'd expect this method to call each_source, and we just refactor the guts of each_source to be tenant aware.